utils/python/CIME/SystemTests/system_tests_common.py

```diff
@@ -30,9 +30,23 @@ def __init__(self, case, expected=None):
         self._casebaseid = self._case.get_value("CASEBASEID")
         self._test_status = TestStatus(test_dir=caseroot, test_name=self._casebaseid)
 
+        self._init_environment(caseroot)
+        self._init_locked_files(caseroot, expected)
+        self._init_case_setup()
```

Comment:
I needed to break out these interactions with the environment and case object into separate methods so that I could override them for unit tests (avoiding their operation). I could have just split out one method, but I figured I'd give helpful names to the different things done here while I was at it.